### PR TITLE
refactor: drop IE < 9 support

### PR DESCRIPTION
**Pull request by shvaikalesh**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -20,8 +17,8 @@ var symbolIteratorExists = symbolExists && typeof Symbol.iterator !== 'undefined
 var symbolToStringTagExists = symbolExists && typeof Symbol.toStringTag !== 'undefined';
 var setEntriesExists = setExists && typeof Set.prototype.entries === 'function';
 var mapEntriesExists = mapExists && typeof Map.prototype.entries === 'function';
-var setIteratorPrototype = getPrototypeOfExists && setEntriesExists && Object.getPrototypeOf(new Set().entries());
-var mapIteratorPrototype = getPrototypeOfExists && mapEntriesExists && Object.getPrototypeOf(new Map().entries());
+var setIteratorPrototype = setEntriesExists && Object.getPrototypeOf(new Set().entries());
+var mapIteratorPrototype = mapEntriesExists && Object.getPrototypeOf(new Map().entries());
 var arrayIteratorExists = symbolIteratorExists && typeof Array.prototype[Symbol.iterator] === 'function';
 var arrayIteratorPrototype = arrayIteratorExists && Object.getPrototypeOf([][Symbol.iterator]());
```
Comment (shvaikalesh) [inline index.js]: Please, notice getPrototypeOfExists && missing here: IE8 would fail on this.

File: index.js [hunk 2]
```diff
@@ -159,7 +156,7 @@ module.exports = function typeDetect(obj) {
      * Test: `Object.prototype.toString.call(document.createElement('blockquote'))``
      *  - IE <=10 === "[object HTMLBlockElement]"
      */
-    if (htmlElementExists && obj instanceof HTMLElement && obj.tagName === 'BLOCKQUOTE') {
```
Comment (shvaikalesh) [inline index.js]: Having `location` and `document` we can be kinda sure `HTMLElement` is there.

**Comment by meeber** [comment]
Removing `getPrototypeOfExists` seems reasonable.

I don't know know enough about `HTMLElement` to know if there are any gotchas there, or if that check existed there to address a known issue. Will defer to @keithamus.

As for the indentation fixes: In general, I think it's better to separate large scale linting fixes from specific issue PRs like this one. Or at least move the linting changes to a specific commit. Makes it easier to review and also if there's a problem down the road that requires reverting the specific-issue-fix, it doesn't also revert the linting changes.

**Comment by shvaikalesh** [comment]
Thanks for the review, @meeber. Removal of `if (getPrototypeOfExists) {` branch caused bad diff, no linting fixes there.

**Comment by meeber** [comment]
Whoops, I see now. Disregard!

**Comment by keithamus** [comment]
Okay, this all looks completely reasonable to me. However, I would ask that we add `BREAKING CHANGE:` to the commit message because we're dropping compat to a browser - while that browser is not documented as supported, we should be nice and follow principle of least surprise and not drop support on patch/minor. Thoughts?

**Comment by shvaikalesh** [comment]
@keithamus yep, let's do `breaking` this. Not so long ago, uglify dropped IE8 support by default in non-major release and it was not well received by the community.

**Comment by keithamus** [comment]
Whelp, LGTM 👍 😄

**Comment by keithamus** [comment]
@meeber @lucasfcosta whenever you get a chance to (re)review this, that'd be sweeeet

**Comment by meeber** [comment]
LGTM

**Pull request merged**